<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1075</width>
    <height>699</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Computer Graphics Lab</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>801</width>
      <height>601</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout"/>
   </widget>
   <widget class="QPushButton" name="openObjFilePushButton">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>610</y>
      <width>799</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Open .obj file</string>
    </property>
   </widget>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>50</y>
      <width>251</width>
      <height>581</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="page_3">
     <widget class="QGroupBox" name="animationGroupBox">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>11</y>
        <width>142</width>
        <height>192</height>
       </rect>
      </property>
      <property name="title">
       <string>Animations</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QRadioButton" name="noAnimationRadioButton">
         <property name="text">
          <string>No animation</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="xRotationRadioButton">
         <property name="text">
          <string>X rotation</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="yRotationRadioButton">
         <property name="text">
          <string>Y rotation</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="zRotationRadioButton">
         <property name="text">
          <string>Z rotation</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="carouselRadioButton">
         <property name="text">
          <string>Carousel</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>220</y>
        <width>37</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>FOVy:</string>
      </property>
     </widget>
     <widget class="QSlider" name="fovySlider">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>250</y>
        <width>181</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>30</number>
      </property>
      <property name="maximum">
       <number>120</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>290</y>
        <width>135</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Near plane distance:</string>
      </property>
     </widget>
     <widget class="QLabel" name="nearPlaneDistanceLabel">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>290</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="nearPlaneDistanceSlider">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>320</y>
        <width>181</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>400</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>360</y>
        <width>122</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Far plane distance:</string>
      </property>
     </widget>
     <widget class="QLabel" name="farPlaneDistanceLabel">
      <property name="geometry">
       <rect>
        <x>143</x>
        <y>360</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>500</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="farPlaneDistanceSlider">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>390</y>
        <width>181</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>500</number>
      </property>
      <property name="sliderPosition">
       <number>500</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>430</y>
        <width>149</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Orbit camera distance:</string>
      </property>
     </widget>
     <widget class="QLabel" name="orbitCameraDistanceLabel">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>430</y>
        <width>24</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>200</string>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider">
      <property name="geometry">
       <rect>
        <x>11</x>
        <y>460</y>
        <width>181</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>5</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="value">
       <number>200</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>220</y>
        <width>53</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>degrees</string>
      </property>
     </widget>
     <widget class="QLabel" name="fovyDegreesValueLabel">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>220</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>30</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <widget class="QCheckBox" name="drawWorldAxesCheckBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>141</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Draw world axes</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="drawPolygonMeshCheckBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>161</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Draw polygon mesh</string>
      </property>
     </widget>
     <widget class="QLabel" name="penColorLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 0);
border: 1px solid black;</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="changePenColorButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>90</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Choose Pen Color</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="rasterizePolygonsCheckBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>151</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Rasterize polygons</string>
      </property>
     </widget>
     <widget class="QLabel" name="brushColorLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(127, 127, 127);
border: 1px solid black;</string>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="changeBrushColorButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>190</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Choose Brush Color</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="enableZBufferingCheckBox">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>251</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Enable Z-buffering (W-buffering)</string>
      </property>
     </widget>
     <widget class="QLabel" name="meshStatusLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>221</width>
        <height>71</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(200, 200, 200);
                    color: black;
                    border: 1px solid black;
                    font-weight: bold</string>
      </property>
      <property name="text">
       <string>Mesh Status displayed here</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="modifyMeshButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>310</y>
        <width>93</width>
        <height>29</height>
       </rect>
      </property>
      <property name="text">
       <string>Modify</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="enableBackfaceCullingCheckBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>380</y>
        <width>251</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Enable backface culling</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page">
     <widget class="QLabel" name="normalStatusLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>300</y>
        <width>221</width>
        <height>71</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(200, 200, 200);
                    color: black;
                    border: 1px solid black;
                    font-weight: bold</string>
      </property>
      <property name="text">
       <string>Normal Status displayed here</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="lightPositionSlider">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>221</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>360</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Light position:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>30</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>degrees</string>
      </property>
     </widget>
     <widget class="QLabel" name="lightPositionDegreesValueLabel">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>31</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="enableLambertianModelCheckBox">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>221</width>
        <height>24</height>
       </rect>
      </property>
      <property name="text">
       <string>Enable Lambertian Model</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QComboBox" name="guiPageSelection">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>15</y>
      <width>241</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1075</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
